swiss_locator/core/filters/swiss_locator_filter_vector_tiles.py

```diff
@@ -0,0 +1,86 @@
+from qgis.gui import QgisInterface
+from qgis.core import (
+    QgsApplication,
+    QgsBlockingNetworkRequest,
+    QgsFetchedContent,
+    QgsLocatorResult,
+    QgsFeedback,
+)
+from swiss_locator.core.filters.swiss_locator_filter import (
+    SwissLocatorFilter,
+)
+from swiss_locator.core.filters.filter_type import FilterType
+from swiss_locator.core.results import VectorTilesLayerResult
+
+
+class SwissLocatorFilterVectorTiles(SwissLocatorFilter):
+    def __init__(self, iface: QgisInterface = None, crs: str = None):
+        super().__init__(FilterType.VectorTiles, iface, crs)
+
+    def clone(self):
+        return SwissLocatorFilterVectorTiles(crs=self.crs)
+
+    def displayName(self):
+        return self.tr("Swiss Geoportal Vector Tile Layers")
+
+    def prefix(self):
+        return "chb"
+
+    def hasConfigWidget(self):
+        return False
+
+    def perform_fetch_results(self, search: str, feedback: QgsFeedback):
+        data = {
+            "base map": {
+                "title": "Base map",
+                "description": "",
+                "url": "https://vectortiles.geo.admin.ch/tiles/ch.swisstopo.base.vt/v1.0.0/{z}/{x}/{y}.pbf",
+                "style": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.basemap.vt/style.json"
+            },
+            "light base map": {
+                "title": "Light base map", "description": "",
+                "url": "https://vectortiles.geo.admin.ch/tiles/ch.swisstopo.base.vt/v1.0.0/{z}/{x}/{y}.pbf",
+                "style": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.lightbasemap.vt/style.json"
+            },
+            "imagery base map": {
+                "title": "Imagery base map", "description": "",
+                "url": "https://vectortiles.geo.admin.ch/tiles/ch.swisstopo.base.vt/v1.0.0/{z}/{x}/{y}.pbf",
+                "style": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.imagerybasemap.vt/style.json"
+            },
+            "leichte-basiskarte": {
+                "title": "leichte-basiskarte", "description": "",
+                "url": "https://vectortiles.geo.admin.ch/tiles/ch.swisstopo.leichte-basiskarte.vt/v3.0.1/{z}/{x}/{y}.pbf",
+                "style": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte.vt/style.json"
+            },
+            "leichte-basiskarte-imagery": {
+                "title": "leichte-basiskarte-imagery", "description": "",
+                "url": "https://vectortiles.geo.admin.ch/tiles/ch.swisstopo.leichte-basiskarte.vt/v3.0.1/{z}/{x}/{y}.pbf",
+                "style": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.leichte-basiskarte-imagery.vt/style.json"
+            },
+        }
+
+        for keyword in list(data.keys()):
+            results = {}
+            score = 1
+            if search.lower() in keyword:
```

Comment:
```suggestion
            if search.lower() in keyword:
```

If no text is typed (or less than 3 chars), I would show all the results. The idea is that if you type the prefix `chb` or click on the filter from the locator widget, you'd like to see all base maps listed without actually searching for it. Does this make sense?

Reply:
I agree! Much better for users.

Done in https://github.com/opengisch/qgis-swiss-locator/pull/79/commits/4631f047cb176a777ecc64c2233dbb8accadddf4

Now, either by selecting the filter prefix or by entering it (`chb` + blank space), users will get all available base maps listed. Searches are also supported.